packages/editor/src/components/rich-text/index.js

```diff
@@ -1059,6 +1059,17 @@ RichTextContainer.Content = ( { value, format, tagName: Tag, ...props } ) => {
 	return content;
 };
 
+/**
+ * Check if the given `RichText` value is empty on not.
+ *
+ * @param {Array} value `RichText` value.
+ *
+ * @return {boolean} True if empty, false if not.
+ */
+RichTextContainer.isEmpty = ( value ) => {
+	return ! value || ! value.length;
```

Comment:
If this is just whitespace, do we want to consider it empty? It feels weird to have a bunch of newlines be considered "content", but I guess that's historically what we do so we should keep it that way, right?

It just seems odd when a bunch of newlines make captions appear:

<img width="713" alt="screenshot 2018-08-28 13 10 14" src="https://user-images.githubusercontent.com/90871/44722053-bfeb1780-aac3-11e8-83e8-1fe97be50ded.png">

Reply:
> It just seems odd when a bunch of newlines make captions appear:

Do you mean disappear? Yeah, I think we should consider it not empty if there are newline, as there is "something" in it and you can still move the caret and press backspace.

Reply:
Ah, I mean that in that example the caption shows up but it's just newlines. But that's how it is here and in `master`:

![2018-08-28 14 38 14](https://user-images.githubusercontent.com/90871/44726631-0f374500-aad0-11e8-8d67-998b4ebb27b3.gif)

But like you say: that makes sense since you can move the caret around.

Reply:
Yeah, that's maybe a bit weird but totally expected. Maybe that's something we can address later on in an effort to visualise invisible characters like non breaking spaces, line breaks, etc.